controllers/serviceaccount_controller.go

```diff
@@ -0,0 +1,111 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	v1 "alexellis/registry-creds/api/v1"
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ServiceAccountReconciler reconciles a ServiceAccount object
+type ServiceAccountReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=core,resources=serviceaccounts,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core,resources=serviceaccounts/status,verbs=get;update;patch
+
+func (r *ServiceAccountReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
```

Comment:
We could probably make do with a watcher vs a reconciler.

As per -> https://github.com/alexellis/registry-creds/blob/master/controllers/namespace_watcher.go

This may just be a naming thing?

Reply:
Yep, it's just a naming thing, we can change it as serviceaccount_watcher.

Reply:
done 216d573.